index.html

```diff
@@ -259,10 +259,200 @@ <h1>Architecture</h1>
+		<li><span class="rfc2119-assertion" id="consumer-merge-intros">
+			A D-Client MUST be able to merge URLs resulting from 
+			multiple Introduction mechanisms, multiple results from a single Introduction mechanism,
+			and multiple Introduction invocations into a single set.</span>
+			Note that even if only one Introduction mechanism is supported multiple results might be 
+			produced.  This assertion states that the overall output of the Introduction phase is
+			a single set of URLs.  
+			The word "set" is used here in the mathematical sense: the results are unordered.
+			To clarify, if the D-Client only supports one Introduction mechanism and that mechanism
+			can only produce one URL, then the "merge" is trivial.
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-td-identify">A D-Client MUST be able to identify whether a TD fetched from an 
+			Introduction URL is a ThingDirectory or a ThingLink.</span>
+			This does not imply that the D-Client needs to take any action on such TDs; it just needs to be able
+			to check the <code>@type</code> field and make this distinction.
+                        A D-Client can decide whether or not to follow links or fetch TDD contents.
+	                There are some use cases where a Consumer may not expand certain URLs, for example links pointing
+	                at external resources, or when a TDD contains many TDs and fetching them all
+	                would exceed the Consumer's memory processing capabilities.
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-fetch-tdd">A D-Client MAY fetch additional TDs from any Exploration mechanism
+			described in its initial set of TDs (including, in particular, Thing Description Directories) 
+			and add them into the set of TD results.</span>
+			This only adds the results of fetching TDs from a Exploration mechanism to the set of
+			results. These new results do not delete the original TD describing the Exploration mechanism. 
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-fetch-links">A D-Client MAY fetch source TDs from the targets of the links in a Thing Link
+			described in its initial set of TDs and add them into the set of TD results.</span>
+			This only adds the results of fetching TDs from a Thing Link to the set of
+			results. These new results do not delete the original TD describing the Exploration mechanism. 
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-fetch-iteration">A D-Client MAY fetch additional TDs iteratively from any Thing Link or
+			Exploration mechanism described in its set of TDs and add them into the set of TD results.</span>
+			This only adds the results of fetching TDs from a Thing Link to the set of
+			results. These new results do not delete the original TD describing the Exploration mechanism. 
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-termination">A D-Client MAY terminate fetching additional TDs at any point or for
+			any reason.</span>
```

Comment:
I actually have a specific assertion that lets a Discoverer stop for any reason (for instance, they could simply have hit a time limit or a memory limit).  This will make discovery a bit inconsistent between implementations, but it would be anyway given the other variations allowed (e.g. types of intros).